sdk/iot/Azure.Iot.Hub.Service/src/IoTHubServiceClient.cs

```diff
@@ -54,24 +61,42 @@ protected IoTHubServiceClient()
         /// <summary>
         /// Initializes a new instance of the <see cref="IoTHubServiceClient"/> class.
         /// </summary>
-        public IoTHubServiceClient(Uri endpoint)
-            : this(endpoint, new IoTHubServiceClientOptions())
+        /// <param name="connectionString">
+        /// The IoT Hub connection string, with either "iothubowner", "service", "registryRead" or "registryReadWrite" policy, as applicable.
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#access-control-and-permissions"/>.
+        /// </param>
+        public IoTHubServiceClient(string connectionString)
+            : this(connectionString, new IoTHubServiceClientOptions())
         {
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="IoTHubServiceClient"/> class.
         /// </summary>
-        public IoTHubServiceClient(Uri endpoint, IoTHubServiceClientOptions options)
+        /// <param name="connectionString">
+        /// The IoT Hub connection string, with either "iothubowner", "service", "registryRead" or "registryReadWrite" policy, as applicable.
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#access-control-and-permissions"/>.
+        /// </param>
+        /// <param name="options">
+        /// Options that allow configuration of requests sent to the IoT Hub service.
+        /// </param>
+        public IoTHubServiceClient(string connectionString, IoTHubServiceClientOptions options)
         {
             Argument.AssertNotNull(options, nameof(options));
 
+            var iotHubConnectionString = new IotHubConnectionString(connectionString);
+            ISasTokenProvider sasProvider = iotHubConnectionString.GetSasTokenProvider();
+
+            _endpoint = new UriBuilder { Scheme = "https", Host = iotHubConnectionString.HostName, }.Uri;
```

Comment:
This could be in a helper method or in IotHubConnectionString.

Reply:
done.